app/src/playStore/java/org/fossasia/openevent/general/search/GeoLocationUI.kt

```diff
@@ -0,0 +1,82 @@
+import com.google.android.gms.location.LocationServices
+import kotlinx.android.synthetic.main.activity_search_location.*
+import org.fossasia.openevent.general.MainActivity
+import timber.log.Timber
+import java.io.IOException
+import java.util.*
+
+private const val TO_SEARCH: String = "ToSearchFragment"
+
+class GeoLocationUI {
+
+    fun configure(activity: Activity, searchLocationViewModel: SearchLocationViewModel) {
+        val permission = ContextCompat.checkSelfPermission(activity, Manifest.permission.ACCESS_COARSE_LOCATION)
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(activity, arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST)
+            return
+        }
+        val service = activity.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+        val enabled = service.isProviderEnabled(LocationManager.GPS_PROVIDER)
+        if (!enabled) {
+            val intent = Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS)
+            activity.startActivity(intent)
+        }
+        activity.currentLocation.setOnClickListener {
+            val locationRequest: LocationRequest = LocationRequest.create()
+            locationRequest.priority = LocationRequest.PRIORITY_LOW_POWER
+            val locationCallback = object : LocationCallback() {
+                override fun onLocationResult(locationResult: LocationResult?) {
+                    if (locationResult == null) {
+                        return
+                    }
+                    for (location in locationResult.locations) {
+                        if (location != null) {
+                            val latitude = location.latitude
+                            val longitude = location.longitude
+                            try {
+                                val geocoder = Geocoder(activity, Locale.getDefault())
+                                val addresses: List<Address> = geocoder.getFromLocation(latitude, longitude, 2)
+                                for (address: Address in addresses) {
+                                    if (address.locality != null) {
+                                        search(activity, SearchLocationActivity().fromSearchFragment, searchLocationViewModel, address.locality)
+                                    }
+                                }
+                            } catch (exception: IOException) {
+                                Timber.e(exception, "Error Fetching Location")
+                            }
+                        }
+                    }
+                }
+            }
+            LocationServices.getFusedLocationProviderClient(activity).requestLocationUpdates(locationRequest, locationCallback, null)
+        }
+    }
+
+    fun search(activity: Activity, fromSearchFragment: Boolean, searchLocationViewModel: SearchLocationViewModel, query: String) {
```

Comment:
Where os this used?

Reply:
inside LocationResult when location is found

Reply:
Line 58